### PR TITLE
[tech] Initial upload of the 'skip_fail' macro

### PR DESCRIPTION
**Pull request by woshilapin**

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+[package]
+name = "skip-fail"
+version = "1.0.0"
+license = "MIT"
+authors = ["Jean SIMARD <woshilapin@tuziwo.info>"]
```
Comment (datanel) [inline Cargo.toml]: I would rather put `Kisio Digital <team.coretools@kisio.org>`

File: src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+//!   let number: u32 = skip_fail!(string_number.parse());
+//! }
+//! # }
+//! ```
+//!
+//! If you want the error to be logged, you can use the feature `log`. The
+//! logging will be done in WARN level with the standard logging interface
+//! provided by [`log`](https://crates.io/crates/log).
+#[macro_export]
+#[cfg(not(feature = "log"))]
```
Comment (datanel) [inline src/lib.rs]: interesting but I think it would better if we can use the 2 macros at the sam time

File: Cargo.toml (hunk 1 above)
Comment (woshilapin) [reply Cargo.toml]: Indeed, I just created the project with `cargo new` and I didn't even touch some of these properties. But you proposition is much better!
Done in 92607094ab55e04e85457146ca6834b790aa1166.

File: src/lib.rs (hunk 2 above)
Comment (woshilapin) [reply src/lib.rs]: I like this idea. It didn't even crossed my mind. I see 2 ways to achieve this functionality:
- adding a parameter to the macro (a boolean `with_log`)
- creating 2 macros named differently (we could name it `skip_fail_with_warn`)

I'm much more in favor of the second solution (we could even think of a third one later with `skip_fail_with_info`).

What do you think?

Comment (datanel) [reply src/lib.rs]: I prefer `skip_fail_with_warn` but a bit different
WHy not something like `skip_fail_and_log` with a second optional parameter `log_level`?

Comment (woshilapin) [reply src/lib.rs]: I'm fine with that, I'll do that.

Comment (woshilapin) [reply src/lib.rs]: I've added the macro `skip_fail_and_log!()` under the feature `log`. This macro now takes 2 parameters, the `Result` and a `log::Level` (which is an enum). I also added more documentation, for each macro.

File: src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+        match $result {
+            Ok(value) => value,
+            Err(error) => {
+                match $log_level {
+                    log::Level::Info => log::info!("{}", error),
+                    log::Level::Warn => log::warn!("{}", error),
+                    log::Level::Error => log::error!("{}", error),
+                    log::Level::Debug => log::debug!("{}", error),
+                    log::Level::Trace => log::trace!("{}", error),
+                }
```
Comment (datanel) [inline src/lib.rs]: I think you can do something like
```suggestion
                log::log($log_level, "{}", error);
```

Comment (woshilapin) [reply src/lib.rs]: Much better indeed. Done in f9c8191d1d803e2c60b9e49d166eec44192af29d.

**Review by datanel — APPROVED** [review]

**Pull request merged**